pkg/reconciler/internal/updater/updater_test.go

```diff
@@ -177,9 +184,85 @@ var _ = Describe("Updater", func() {
 			Expect(found).To(BeTrue())
 			Expect(err).To(Succeed())
 		})
+
+		It("should add a finalizer", func() {
+			u.Update(func(u *unstructured.Unstructured) bool {
+				return controllerutil.AddFinalizer(u, testFinalizer)
+			})
+			Expect(u.Apply(context.TODO(), obj)).To(Succeed())
+			Expect(cl.Get(context.TODO(), types.NamespacedName{Namespace: "testNamespace", Name: "testDeployment"}, obj)).To(Succeed())
+			Expect(obj.GetFinalizers()).To(ContainElement(testFinalizer))
+		})
+
+		It("should remove a finalizer", func() {
+			obj.SetFinalizers([]string{testFinalizer})
+			Expect(cl.Update(context.TODO(), obj)).To(Succeed())
+
+			u.Update(func(u *unstructured.Unstructured) bool {
+				return controllerutil.RemoveFinalizer(u, testFinalizer)
+			})
+			Expect(u.Apply(context.TODO(), obj)).To(Succeed())
+			Expect(cl.Get(context.TODO(), types.NamespacedName{Namespace: "testNamespace", Name: "testDeployment"}, obj)).To(Succeed())
+			Expect(obj.GetFinalizers()).ToNot(ContainElement(testFinalizer))
+		})
+
+		It("should preserve unknown status conditions", func() {
+			// Add external status condition on cluster.
+			clusterObj := obj.DeepCopy()
+			unknownCondition := map[string]interface{}{
+				"type":   "UnknownCondition",
+				"status": string(corev1.ConditionTrue),
+				"reason": "ExternallyManaged",
+			}
+			Expect(unstructured.SetNestedSlice(clusterObj.Object, []interface{}{unknownCondition}, "status", "conditions")).To(Succeed())
+			Expect(retryOnTransientError(func() error {
+				return cl.Status().Update(context.TODO(), clusterObj)
+			})).ToNot(HaveOccurred())
```

Comment:
This reads funny. Perhaps instead:

```suggestion
			err := retryOnTransientError(func() error {
				return cl.Status().Update(context.TODO(), clusterObj)
			})
			Expect(err).ToNot(HaveOccurred())
```

Reply:
done